aiscripts/mule.lib.evaluate_tradeoffers.xml

```diff
@@ -0,0 +1,113 @@
+<?xml version="1.0" encoding="iso-8859-1" ?>
+<aiscript name="mule.lib.evaluate_tradeoffers" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://x4libonline.ddns.net\libraries\aiscripts.xsd" version="1">
+	<!-- Compare tradeoffers by profit and return the most profitable one -->
+	<params>
+		<!-- list of possible buyoffers -->
+		<param name="buyoffers"/>
+		<!-- list of possible selloffers -->
+		<param name="selloffers" />
+		<!-- shipEntity that performes the trade. -->
+		<!-- determines the account used to perform trades -->
+		<!-- determines the cargo size -->
+		<param name="shipEntity"/>
+		<!-- list of already used selloffers -->
+		<param name="usedSelloffers"/>
+		<!-- player buy modifier to adjust supply prices -->
+		<param name="sameFactionBuyMod"/>
+		<!-- current amount of cargo reserved for other wares -->
+		<param name="occupiedCargo"/>
+		<!-- minimum cargo size for allowed transport -->
+		<param name="minCargoSize"/>
+		<!-- Enable trading for build storage -->
+		<!-- if the trade will finish off the build storage for that good it will be allowed for low volumes -->
+		<param name="tradeForBuildStorage" default="false"/>
+		<!-- enable/disable debug messages -->
+		<param name="debugchance"/>
+		<param name="debugFileName"/>
+		<param name="debugDirName"/>
+  </params>
+  <attention min="unknown">
+    <actions>
+			<!-- Default values for best trade -->
+			<set_value name="$bestProfit" exact="0" />
+			<set_value name="$bestAmount" exact="0" />
+			<set_value name="$bestNeedTrade" exact="null" />
+			<set_value name="$bestSupplyTrade" exact="null" />
+
+			<debug_to_file chance="$debugchance" name="$debugFileName" directory="$debugDirName" text="'----- trade offer comparisons'" />
+			<!-- nested loop(s) over the trade offers -->
+			<do_all exact="$buyoffers.count" counter="$i">
+				<set_value name="$someNeed" exact="$buyoffers.{$i}" />
+				<do_all exact="$selloffers.count" counter="$j">
+					<set_value name="$someSupply" exact="$selloffers.{$j}" />
+					
+					<!-- ignores supply side trades already found -->
+					<do_if value="$usedSelloffers.indexof.{$someSupply}">
+						<continue />
+					</do_if>
+					<!-- only do any work if the offers are for the same ware -->
+					<do_if value="$someNeed.ware.id != $someSupply.ware.id">
+						<continue />
+					</do_if>
+
+					<!-- how much does the need want? -->
+					<get_ware_reservation object="$someNeed.owner" type="buy" ware="$someNeed.ware" result="$needReservations" />
+					<set_value name="$targetOfferedAmount" exact="$someNeed.desiredamount" />
```

Comment:
In the original script there was a different value here depending on whether the buyer was a player-owned object or AI owned. I used $someNeed.amount in the AI case. It's not clear whether this distinction is still needed with clamping now in place.

Reply:
Form my reading of the docs and the experience with clamping trade, clamping trade should be enough here. 
But we should confirm this.